qiskit/transpiler/passes/layout/enlarge_with_ancilla.py

```diff
@@ -41,7 +41,7 @@ def run(self, dag):
         if layout is None:
             raise TranspilerError('EnlargeWithAncilla requires property_set["layout"]')
 
-        new_qregs = {reg for reg in layout.get_registers() if reg not in dag.qregs.values()}
+        new_qregs = {reg for reg in layout._regs if reg not in dag.qregs.values()}
```

Comment:
```suggestion
        new_qregs = {reg for reg in layout.get_registers() if reg not in dag.qregs.values()}
```
Only a single call, outside a loop, so is the public API okay here?